ProcessLib/ComponentTransport/ComponentTransportFEM.h

```diff
@@ -131,48 +151,116 @@ class LocalAssemblerData : public ComponentTransportLocalAssemblerInterface
         auto local_b = MathLib::createZeroedVector<LocalVectorType>(
             local_b_data, local_matrix_size);
 
+        // Get block matrices
+        auto Kpp = local_K.template block<pressure_size, pressure_size>(
+            pressure_index, pressure_index);
+        auto Mpp = local_M.template block<pressure_size, pressure_size>(
+            pressure_index, pressure_index);
+        auto Bp = local_b.template segment<pressure_size>(pressure_index);
+
+        auto local_p = Eigen::Map<const NodalVectorType>(
+            &local_x[pressure_index], pressure_size);
+
+        // Nodal DOFs include pressure
+        auto const number_of_components = num_nodal_dof - 1;
+        for (int component_id = 0; component_id < number_of_components;
+             ++component_id)
+        {
+            /*  Partitioned assembler matrix
+             *  |  pp | pc1 | pc2 | pc3 |
+             *  |-----|-----|-----|-----|
+             *  | c1p | c1c1|  0  |  0  |
+             *  |-----|-----|-----|-----|
+             *  | c2p |  0  | c2c2|  0  |
+             *  |-----|-----|-----|-----|
+             *  | c3p |  0  |  0  | c3c3|
+             */
+            auto concentration_index =
+                pressure_size + component_id * concentration_size;
+
+            auto KCC =
+                local_K.template block<concentration_size, concentration_size>(
+                    concentration_index, concentration_index);
+            auto MCC =
+                local_M.template block<concentration_size, concentration_size>(
+                    concentration_index, concentration_index);
+            auto MCp =
+                local_M.template block<concentration_size, pressure_size>(
+                    concentration_index, pressure_index);
+            auto MpC =
+                local_M.template block<pressure_size, concentration_size>(
+                    pressure_index, concentration_index);
+
+            auto local_C = Eigen::Map<const NodalVectorType>(
+                &local_x[concentration_index], concentration_size);
+
+            assembleBlockMatrices(component_id, t, local_C, local_p, KCC, MCC,
+                                  MCp, MpC, Kpp, Mpp, Bp);
+        }
+    }
+
+    void assembleBlockMatrices(
```

Comment:
Please, clang format the complete function signature.

Reply:
I rerun clang-format. There is no changes in function signature, though.

Reply:
Seems to be a bug in clang format. If you put all the code of the function signature in one line and rerun clang format the result will change.

Reply:
It works! I get the right clang-format outputs.